experimental/framework/fluid-static/src/types.ts

```diff
@@ -71,3 +71,70 @@ export interface ContainerSchema {
      */
     dynamicObjectTypes?: LoadableObjectClass<any>[];
 }
+
+/**
+ * Base interface to be implemented to fetch each service's audience. The generic C allows consumers to further
+ * extend the client object with service-specific details about the connecting client, such as device information,
+ * environme
+ */
+export interface IServiceAudience<M extends IMember> {
+    /**
+     * Returns an map of all users currently in the Fluid session where key is the userId and the value is the
+     * member object
+     */
+    getMembers(): Map<string, M>;
+
+    /**
+     * Returns the current active user on this client once they are connected. Otherwise, returns undefined.
+     */
+    getMyself(): M | undefined;
+
+    /**
+     * Returns the current client's details once it is connected. Otherwise, returns undefined.
+     */
+    getMyClient(): IConnectedClient | undefined;
+
+    /**
+     * Collection of event listeners that trigger if the members in the Fluid session change and if a member makes an
+     * edit which updates when the last change was made
+     */
+    on(event: "membersChanged", listener: (members: M[]) => void): this;
+    on(event: "lastEditedChanged", listener: (result: ILastEditedResult<M>) => void): this;
+    off(event: "membersChanged", listener: (members: M[]) => void): this;
+    off(event: "lastEditedChanged", listener: (result: ILastEditedResult<M>) => void): this;
+    once(event: "membersChanged", listener: (members: M[]) => void): this;
+    once(event: "lastEditedChanged", listener: (result: ILastEditedResult<M>) => void): this;
+}
+
+/**
+ * Base interface for providing client information for each connection made to the Fluid session, which will be
+ * different even if it is by the same user, i.e. the clientId will be uniquely generated for each time the user
+ * connects and each indvidiual connection will track its own time that it was last active at. This interface
+ * can be extended to provide additional information specific to each service.
+ */
+export interface IConnectedClient {
+    clientId: string;
+    connectionMode: "write" | "read";
+    timeLastActive?: Date;
```

Comment:
I don't think we should internalize the last edited feature to the audience.  If the audience is a solid representation of connected clients then any metadata we want to track about those clients can be implemented as a standalone structure that observes the audience.  Internalizing metadata features invites future scope creep to the audience and also isn't needed by many customers.

For this PR I'd recommend just omitting the last edited functionality completely, and introduce it in a followup.

Reply:
Last edited functionality has now been completely removed